File: back/src/main/java/com/back/global/security/jwt/TokenHashUtil.java
package com.back.global.security.jwt;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.util.HexFormat;

public class TokenHashUtil {

    public static String sha256(String token) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hash = digest.digest(token.getBytes(StandardCharsets.UTF_8));

            return HexFormat.of().formatHex(hash);
        } catch (Exception e) {
            throw new RuntimeException("토큰 해싱 실패", e);
        }
    }
}

File: back/src/main/kotlin/com/back/global/security/jwt/TokenHashUtil.kt
package com.back.global.security.jwt

import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.util.HexFormat

object TokenHashUtil {
    fun sha256(token: String): String {
        try {
            val digest = MessageDigest.getInstance("SHA-256")
            val hash = digest.digest(token.toByteArray(StandardCharsets.UTF_8))

            return HexFormat.of().formatHex(hash)
        } catch (e: Exception) {
            throw RuntimeException("토큰 해싱 실패", e)
        }
    }
}
